File: javascript/test/Behavioral/Memento.js
(function() {
  define(['Behavioral/Memento/DvdDetails', 'Behavioral/Memento/DvdMemento'], function(DvdDetails, DvdMemento) {
    'use strict';
    return describe('Test Behavioral Memento', function() {
      var dvdDetails, dvdMementoCaretaker, stars;
      dvdMementoCaretaker = null;
      stars = ["Guy Pearce"];
      dvdDetails = new DvdDetails("Memento", stars, '1');
      dvdMementoCaretaker = dvdDetails.createDvdMemento();
      it("as first instantiated", function() {
        return expect(dvdDetails.formatDvdDetails()).toEqual('DVD: Memento starring: Guy Pearce encoding region: 1');
      });
      it('oops - Cappuccino on the keyboard!', function() {
        dvdDetails.addStar("edskdzkvdfb");
        return expect(dvdDetails.formatDvdDetails()).toEqual('DVD: Memento starring: Guy Pearce, edskdzkvdfb encoding region: 1');
      });
      return it('back off changes', function() {
        dvdDetails.setDvdMemento(dvdMementoCaretaker);
        return expect(dvdDetails.formatDvdDetails()).toEqual('DVD: Memento starring: Guy Pearce encoding region: 1');
      });
    });
  });

}).call(this);

File: scala/src/test/scala/Behavioral/Memento.scala
//Memento - testing the Memento - contains a Caretaker object


package test.Behavioral;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Behavioral.Memento.*;


import java.util.ArrayList;

public class Memento {

	@Test
   public void mementotest() {
	    
       DvdDetails.DvdMemento dvdMementoCaretaker;  
       //the Caretaker
       
       ArrayList stars = new ArrayList();
       stars.add(new String("Guy Pearce"));
       DvdDetails dvdDetails = new DvdDetails("Memento", stars, '1'); 
       dvdMementoCaretaker = dvdDetails.createDvdMemento();
       org.junit.Assert.assertEquals("Testing original state", 
			"DVD: Memento, starring: Guy Pearce, encoding region: 1",
			dvdDetails.formatDvdDetails()
		);
       
       dvdDetails.addStar("edskdzkvdfb");  
       org.junit.Assert.assertEquals("Testing after adding star",
			"DVD: Memento, starring: Guy Pearce, edskdzkvdfb, encoding region: 1",
			 dvdDetails.formatDvdDetails()
		);
       
       org.junit.Assert.assertEquals("Memento state", 
			"DVD: Memento, starring: Guy Pearce, encoding region: 1",
			dvdMementoCaretaker.showMemento()
		);
       //show the memento
       
       dvdDetails.setDvdMemento(dvdMementoCaretaker);
       //back off changes
       org.junit.Assert.assertEquals("Showing state revised to reflect memento state", 
			"DVD: Memento, starring: Guy Pearce, encoding region: 1",
			dvdDetails.formatDvdDetails()
		);
		
   }
}      
